SortingMethods/src/SortingMethods/SortMethodsTest.java

```diff
@@ -0,0 +1,101 @@
+package SortingMethods;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import SortingMethods.Shell_Sort;
+import SortingMethods.NaturalComparator;
+import SortingMethods.Counting_Sort;
+
+class SortMethodsTest {
+
+	Shell_Sort<Integer> SH_int;
+	Counting_Sort CS;
+	NaturalComparator<Integer> NaturalComparator;
+
+	@BeforeEach
+	void setUp() {
+		NaturalComparator = new NaturalComparator<Integer>();
+		SH_int = new Shell_Sort<Integer>(NaturalComparator);
+		CS = new Counting_Sort();
+	}
+
+	@Test
```

Comment:
bardzo duzy plus ze jako pierwszy napisales wspolna klase dla testow a nie dla kazdego algorytmu osobna. minus za to, ze testy sa ubogie, malo przypadkow skrajnych i same inty(!)